### PR TITLE
Added the header to zoomed state and created sharableWhenExpanded api endpoint

### PR DESCRIPTION
**Pull request by DeanoDee**
#2

### CODE REVIEW

**Comment by Mackarous** [comment]
This is for issue #2

File: Pod/Classes/ExpandedImageViewController.swift [hunk 1]
```diff
@@ -16,21 +16,36 @@ class ExpandedImageViewController: UIViewController {
...
     // MARK: - Private Outlets
 
     @IBOutlet private weak var scrollView: UIScrollView!
     @IBOutlet private weak var imageView: UIImageView!
     @IBOutlet weak var imageViewAspectRatioConstraint: NSLayoutConstraint!
-
-
+    @IBOutlet var doubleTapGestureRecognizer: UITapGestureRecognizer!
+    @IBOutlet var singleTapGestureRecognizer: UITapGestureRecognizer!
+    @IBOutlet var shareButton: UIBarButtonItem!
```
Comment (Mackarous) [inline Pod/Classes/ExpandedImageViewController.swift]: we should actually make these all private

File: Pod/Classes/ExpandedImageViewController.swift [hunk 2]
```diff
@@ -65,7 +89,29 @@ class ExpandedImageViewController: UIViewController {
         }
     }
 
+    @IBAction func toggleTopBar(sender: AnyObject) {
+        guard let navigationBarHidden = self.navigationController?.navigationBarHidden else {
+            return
+        }
+        self.toggleStatusBar()
+        self.navigationController?.setNavigationBarHidden(!navigationBarHidden, animated: true)
+        UIView.animateWithDuration(Constants.backgroundColorAnimationDuration){
```
Comment (Mackarous) [inline Pod/Classes/ExpandedImageViewController.swift]: space between `)` and `{`

File: Pod/Classes/ExpandedImageViewController.swift [hunk 3]
```diff
@@ -80,6 +126,17 @@ class ExpandedImageViewController: UIViewController {
 
         self.imageView.center = CGPoint(x: x, y: y)
     }
+    
+    private func toggleStatusBar(){
+        self.statusBarHidden = !self.statusBarHidden
+        if self.statusBarHidden {
+             UIView.animateWithDuration(Double(UINavigationControllerHideShowBarDuration)){
```
Comment (Mackarous) [inline Pod/Classes/ExpandedImageViewController.swift]: space between `)` and `{`

**Comment by DeanoDee** [comment]
@a-mehta c'mon let's see some feedback

**Comment by Mackarous** [comment]
Looks good to me. @a-mehta ?

**Pull request merged**